.github/workflows/sqlsmith.yaml

```diff
@@ -0,0 +1,109 @@
+name: SQLsmith
+on:
+  schedule:
+    # run daily 20:00 on master branch
+    - cron: '0 20 * * *'
+  push:
+  pull_request:
+jobs:
+  regress:
+    name: SQLsmith PG${{ matrix.pg }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: ["ubuntu-20.04"]
+        pg: ["14.2"]
+        cc: ["gcc"]
+        build_type: ["Debug"]
+      fail-fast: false
+    env:
+      PG_SRC_DIR: pgbuild
+      PG_INSTALL_DIR: postgresql
+
+    steps:
+    - name: Install Linux Dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install flex bison systemd-coredump gdb clang-9 llvm-9 llvm-9-dev llvm-9-tools build-essential autoconf autoconf-archive libpqxx-dev libboost-regex-dev libsqlite3-dev
+
+    # this workflow depends on the cached postgres build from the main regression
+    # workflow since that workflow runs daily there should always be a cache hit
+    - name: Cache PostgreSQL ${{ matrix.pg }}
+      id: cache-postgresql
+      uses: actions/cache@v2
+      with:
+        path: ~/${{ env.PG_SRC_DIR }}
+        key: ${{ matrix.os }}-postgresql-${{ matrix.pg }}-${{ matrix.cc }}-${{ matrix.build_type }}
+
+    # we abort on cache miss otherwise we would have to reproduce most variables
+    # of the main regression build matrix in this workflow
+    - name: Abort on cache miss
+      if: steps.cache-postgresql.outputs.cache-hit != 'true'
+      run: false
+
+    - name: Install PostgreSQL ${{ matrix.pg }} ${{ matrix.build_type }}
+      run: |
+        make -C ~/$PG_SRC_DIR install
+        make -C ~/$PG_SRC_DIR/contrib/postgres_fdw install
+
+    - name: Checkout TimescaleDB
+      uses: actions/checkout@v2
+
+    - name: Build TimescaleDB
+      run: |
+        ./bootstrap -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} -DPG_SOURCE_DIR=~/$PG_SRC_DIR -DPG_PATH=~/$PG_INSTALL_DIR ${{ matrix.tsdb_build_args }}
+        make -C build
+        make -C build install
+
+    - name: Checkout sqlsmith
+      uses: actions/checkout@v2
+      with:
+        repository: 'svenklemm/sqlsmith'
+        path: 'sqlsmith'
+        ref: 'timescaledb'
```

Comment:
Just a minor comment here... any change to upstream your changes?

Reply:
I think we should at least move it to the Timescale org.

Reply:
some of the changes are timescaledb specific, other than that its mostly trivial/cosmetic stuff so not really worth trying to upstream